### PR TITLE
TEST-#5420: port asv benchmarks for Repr, MaskBool, isNull, dropNa and equals functions

### LINKED ISSUE #5420: [ASV] port Repr, MaskBool, isNull, dropNa and equals benchmarks from pandas github
Have to add ASV benchmarks  which does the following actions

Gets string representation of dataframe
Applies a mask on a dataframe
Detect missing values 
remove missing values
check equality of dataframes

### PR DESCRIPTION
**Pull request by arunjose696**
## What do these changes do?

Added asv benchmarks for Repr, MaskBool, isNull, dropNa and equals functions
Source PR https://github.com/modin-project/modin/pull/5036

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #5420  
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: asv_bench/benchmarks/benchmarks.py [hunk 1]
```diff
@@ -1184,4 +1184,96 @@ def time_series_indices(self, shape):
...
+
+    def setup(self, shape):
+        self.df_no_null = IMPL.DataFrame(np.random.randn(*shape))
+        sample = np.array([np.nan, 1.0])
+        data = np.random.choice(sample, (shape[0], shape[1]))
+        self.df = IMPL.DataFrame(data)
+        execute(self.df), execute(self.df_no_null)
+
+    def time_isnull_floats_no_null(self, shape):
+        execute(IMPL.isnull(self.df_no_null))
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1233]: Let's drop "no_null" case. No big difference for Modin.

```suggestion
        sample = np.array([np.nan, 1.0])
        data = np.random.choice(sample, (shape[0], shape[1]))
        self.df = IMPL.DataFrame(data)
        execute(self.df)
```

File: asv_bench/benchmarks/benchmarks.py [hunk 2]
```diff
@@ -1184,4 +1184,96 @@ def time_series_indices(self, shape):
...
+        self.df_nan.iloc[-1, -1] = np.nan
+        execute(self.df), execute(self.df_nan)
+
+    # returns a  boolean thus not calling execute
+    def time_frame_float_equal(self, shape):
+        self.df.equals(self.df)
+
+    # returns a  boolean thus not calling execute
+    def time_frame_float_unequal(self, shape):
+        self.df.equals(self.df_nan)
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1276]: Let's drop one case for Modin. Having another case is enough to track performance for `equals`.

```suggestion
    def setup(self, shape):
        self.df = IMPL.DataFrame(np.random.randn(*shape))
        self.df.iloc[-1, -1] = np.nan
        execute(self.df)

    # returns a boolean thus not calling execute
    def time_frame_float_unequal(self, shape):
        self.df.equals(self.df_nan)
```

File: asv_bench/benchmarks/benchmarks.py [hunk 3]
```diff
@@ -1184,4 +1184,96 @@ def time_series_indices(self, shape):
...
+
+    def setup(self, how, axis, shape):
+        row, col = shape
+        self.df = IMPL.DataFrame(np.random.randn(row, col))
+        self.df.iloc[row // 20 : row // 10, col // 3 : col // 2] = np.nan
+        self.df_mixed = self.df.copy()
+        self.df_mixed["foo"] = "bar"
+        execute(self.df), execute(self.df_mixed)
+
+    def time_dropna(self, how, axis, shape):
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1252]: Let's drop `time_dropna` case.

File: asv_bench/benchmarks/benchmarks.py [hunk 4]
```diff
@@ -1184,4 +1184,96 @@ def time_series_indices(self, shape):
...
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        df = IMPL.DataFrame(np.random.randn(*shape))
+        df = df.where(df > 0)
+        self.bools = df > 0
+        self.mask = IMPL.isnull(df)
+        execute(self.bools), execute(self.mask)
+
+    def time_frame_mask_bools(self, shape):
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1213]: Let's drop `time_frame_mask_bools` case.

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: asv_bench/benchmarks/benchmarks.py (hunk 4 above)
Comment (arunjose696) [reply asv_bench/benchmarks/benchmarks.py]: dropped the previous time_dropna case and renamed time_dropna_axis_mixed_dtypes to time_dropna

File: asv_bench/benchmarks/benchmarks.py [hunk 5]
```diff
@@ -1184,4 +1184,80 @@ def time_series_indices(self, shape):
...
+class TimeRepr:
+
+    params = [get_benchmark_shapes("TimeRepr")]
+    param_names = ["shape"]
+
+    def setup(self, shape):
+        self.df = IMPL.DataFrame(np.random.randn(*shape))
+        execute(self.df)
+
+    # returns a  string thus not calling execute
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1196]:
```suggestion
    # returns a string thus not calling execute
```

File: asv_bench/benchmarks/benchmarks.py [hunk 6]
```diff
@@ -1184,4 +1184,80 @@ def time_series_indices(self, shape):
...
+class TimeDropna:
+
+    params = (["all", "any"], [0, 1], get_benchmark_shapes("TimeDropna"))
+    param_names = ["how", "axis", "shape"]
+
+    def setup(self, how, axis, shape):
+        row, col = shape
+        self.df = IMPL.DataFrame(np.random.randn(row, col))
+        self.df.iloc[row // 20 : row // 10, col // 3 : col // 2] = np.nan
+        self.df["foo"] = "bar"
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1241]: Why is this column needed here?

File: asv_bench/benchmarks/benchmarks.py [hunk 7]
```diff
@@ -1184,4 +1184,80 @@ def time_series_indices(self, shape):
...
+
+    def setup(self, shape):
+        df = IMPL.DataFrame(np.random.randn(*shape))
+        df = df.where(df > 0)
+        self.bools = df > 0
+        self.mask = IMPL.isnull(df)
+        execute(self.bools), execute(self.mask)
+
+    def time_frame_mask(self, shape):
+        execute(self.bools.astype(float).mask(self.mask))
```
Comment (anmyachev) [inline asv_bench/benchmarks/benchmarks.py:1214]: Why not just like that?
```suggestion
        self.df = IMPL.DataFrame(np.random.randn(*shape))
        self.mask = df < 0
        execute(self.df), execute(self.mask)

    def time_frame_mask(self, shape):
        execute(self.df.mask(self.mask))
```

Comment (arunjose696) [reply asv_bench/benchmarks/benchmarks.py:1214]: this seems better.

File: asv_bench/benchmarks/benchmarks.py (hunk 6 above)
Comment (anmyachev) [reply asv_bench/benchmarks/benchmarks.py:1239]: I don't mind leaving it as it is.

**Review by anmyachev — APPROVED** [review]
LGTM! Thanks @arunjose696

**Pull request merged**